test/metabase/sync/analyze/fingerprint/insights_test.clj

```diff
@@ -89,13 +89,23 @@
                    ["2018-12-02",179,3311]
                    ["2018-12-03",144,2525]])
 
+(defn- round
```

Comment:
I just remembered that we already have a `round-to-decimals` function in `metabase.util`. It's not implemented exactly the same but would it work to use that here instead?

Reply:
That actually does a related but different thing. That rounds to a certain number of _decimal places_; my new function rounds to a certain number of _significant figures_.

The number in question here is 1.567...e227, so rounding to decimal places isn't the issue.

I can move this function into `metabase.util` as `round-to-sig-figures` or something like that?

Reply:
Ah, interesting. Yeah seems useful to have that in `metabase.util` as well. Maybe with brief explanations in the docstrings about what the differences are / when to use each.

Reply:
one thought. might be nice to have some defrecord that uses round-to-sig-figures in its equality check. This could let you use them inline in `partial=`. `(is (partial= {:a (close-enough 3.234e277)}))` or the like. Alternatively, have `partial=` walk the expected form and swap out floats with this close-enough notion maybe?

Reply:
although might be too magical since it wouldn't be able to handle sets. So just kinda spitballin

Reply:
That sort of "matcher" magic is one of the nice features of the old Midje days that I still miss.

I'm moving this new function into `metabase.util` and if we need it in more places we can invest in adding this to `partial=`.

Reply:
I think nubank has some good testing utils in this area as well. But good call on extending as we need and with thought.